### ISSUE TITLE
RFE support building container images with rpms

### ISSUE THREAD

**Issue by vbatts** [bazelbuild/rules_docker#203]
for the `container_image` (or `docker_build`) there is currently no support for rpm based distributions. This is a blocker for a number of use-cases.

**Comment by vbatts** [comment]
I'm open to feedback on how maintainers foresee this would work, but it's something I'm going to have to work on sooner than later.

**Comment by mattmoor** [comment]
@r2d4 @dlorenc @damienmg FYI

**Comment by vbatts** [comment]
@mattmoor ! :wave:  long time, no see!

**Comment by mattmoor** [comment]
Hey @vbatts :)

**Comment by vbatts** [comment]
the rpm file format is not as tidy as an ar archive + tar archive. So subprocess to `rpm2cpio` or something like https://github.com/srossross/rpmfile ?

**Comment by mattmoor** [comment]
I think the library is preferable to shelling out, if it works.  Seems like it has been a while since there's been any activity on that repo.

**Comment by vbatts** [comment]
generally agree. What is the process for adding a pypi dependency?
Also btw, my current WIP is here https://github.com/vbatts/bazel-rules_docker/tree/rpm_support

**Comment by mattmoor** [comment]
tl;dr Use something like [this](https://github.com/bazelbuild/rules_docker/blob/master/container/container.bzl#L82-L86) or [this](https://github.com/bazelbuild/rules_docker/blob/master/container/container.bzl#L106)

I recently added support for PIP in `rules_python`, but the current Bazel `WORKSPACE` model makes it really hard for library writers to use without the transitive `WORKSPACE` boilerplate exploding, so I stopped short of using it here (o/w `rules_docker` users would *have* to do the `pip` dance in their own `WORKSPACE` too)

**Comment by rmohr** [comment]
I guess  this issue can be closed with the decision in #566 in mind?

Like suggested in #566 I created https://github.com/rmohr/rules_container_rpm which contains independent rpm specific rules for containers.

**Comment by vbatts** [comment]
True enough. Thanks @rmohr for doing that work. @mattmoor is there a curated list of these kinds of rules repos available?

**Comment by nlopezgi** [comment]
@vbatts,
no, atm such a list does not officially exist. I think there's conversation happening around this topic in bazel-discuss@googlegroups.com. Unofficially, this list exists: https://github.com/jin/awesome-bazel

**Comment by vbatts** [comment]
@nlopezgi thanks. @rmohr perhaps a PR there could give more visibility

**Comment by rmohr** [comment]
@vbatts thanks for the suggestion. Created a PR there.
